objects/c932e8/script.lua

```diff
@@ -13,42 +40,110 @@ function onLoad()
     end
 end
 
+function onSave()
+    return JSON.encode{
+        initialized = initialized,
+        playerBlock = playerBlock and playerBlock.guid,
+    }
+end
+
 function setupDeepSlumber(_, color)
     doSpiritSetup{color=color}
 end
 
+function findSerpent(color)
+    local elementScanZone = getObjectFromGUID(Global.getVar("elementScanZones")[color])
+    for _, obj in pairs(elementScanZone.getObjects()) do
+        if obj.guid == serpentGUID then
+            return true
+        end
+    end
+    return false
+end
+
 function doSpiritSetup(params)
     local color = params.color
     if not Global.getVar("gameStarted") then
         Player[color].broadcast("Please wait for the game to start before pressing button!", "Red")
         return
     end
 
-    local zone = getObjectFromGUID(Global.getVar("elementScanZones")[color])
-    local objs = zone.getObjects()
-    local found = false
-    local serpent = nil
-    for _,obj in pairs(objs) do
-        if obj.guid == "cebe09" then
-            found = true
-            serpent = obj
-            break
-       end
-    end
-    if not found then
+    if not findSerpent(color) then
         Player[color].broadcast("You have not picked Serpent Slumbering Beneath the Island!", "Red")
         return
     end
 
     local snapPoints = serpent.getSnapPoints()
-
-    table.insert(snapPoints,{position={1.14, 0.20, 0.65}})
-    table.insert(snapPoints,{position={0.92, 0.20, 0.65}})
-    table.insert(snapPoints,{position={0.70, 0.20, 0.65}})
-    table.insert(snapPoints,{position={1.25, 0.20, 0.84}})
-    table.insert(snapPoints,{position={1.03, 0.20, 0.84}})
-    table.insert(snapPoints,{position={0.81, 0.20, 0.84}})
+    for _, position in pairs(slumberPoints) do
+        table.insert(snapPoints, {position = position})
+    end
     serpent.setSnapPoints(snapPoints)
 
-    self.destruct()
+    playerBlock = Global.getVar("playerBlocks")[color]
+
+    initialized = true
+    self.clearButtons()
+    self.locked = true
+    self.interactable = false
+    local position = self.getPosition()
+    position.y = -2
+    self.setPosition(position)
+
+    updatePresence()
+    Wait.time(updatePresence, 1, -1)
+end
+
+local function presenceOnIsland(color)
+    local count = 0
+    for _, presence in pairs(getObjectsWithTag("Presence")) do
+        if presence.getName() == color .. "'s Presence" then
+            local hits = Physics.cast{
+                origin = presence.getBoundsNormalized().center,
+                size = presence.getBoundsNormalized().size,
```

Comment:
so while serpent won't do this, in the world of volcano having a huge tower the size of the bounding box can be quite large. so you may want to set the origin to be the bottom of the box and then do a static sized cast

Reply:
I've updated it to use the bottom center, and reasonable size (though looking at disassembly for an old version, it only looks at the `x` coordinate of the size anyway. It may behave weird for large stacks knocked on their side, but that seems unlikely.